tests/testsuite/help.rs

```diff
@@ -17,9 +15,15 @@ fn help() {
     cargo_process("build -h").run();
     cargo_process("help help").run();
     // Ensure that help output goes to stdout, not stderr.
-    cargo_process("search --help").with_stderr("").run();
+    cargo_process("search --help").with_stderr_data("").run();
     cargo_process("search --help")
-        .with_stdout_contains("[..] --frozen [..]")
+        .with_stdout_data(
```

Comment:
You could use `str![]` and regenerate snapshots by running this command `SNAPSHOTS=overwrite cargo t --test testsuite help::`

```suggestion
        .with_stdout_data(str![])
```

Reply:
Thanks, but somehow I can get `SNAPSHOTS=overwrite` taking effect for other tests, but not here.

<details><summary>Can auto fix other test with <code>SNAPSHOTS=overwrite cargo test --test testsuite binary_name::gated</code></summary>

```
$ SNAPSHOTS=overwrite cargo test --test testsuite binary_name::gated
    Blocking waiting for file lock on build directory
   Compiling cargo v0.82.0 (/Users/ychou/choznerol/contrib/cargo)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 7.24s
     Running tests/testsuite/main.rs (target/debug/deps/testsuite-00971f03e698357e)

running 1 test
Fixing:
---- expected: tests/testsuite/binary_name.rs:34:27
++++ actual:   stderr
   1    1 | [ERROR] failed to parse manifest at `[ROOT]/foo/Cargo.toml`
   2    2 |
   3      - fix me
   4    3 | Caused by:
   5    4 |   feature `different-binary-name` is required
   6    5 |
   7      -   The package requires the Cargo feature called `different-binary-name`, but that feature is not stabilized in this version of Cargo ([..]).
        6 +   The package requires the Cargo feature called `different-binary-name`, but that feature is not stabilized in this version of Cargo (1.81.0 (5f0294b15 2024-06-13)).
   8    7 |   Consider adding `cargo-features = ["different-binary-name"]` to the top of Cargo.toml (above the [package] table) to tell Cargo you are opting in to use this unstable feature.
   9    8 |   See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#different-binary-name for more information about the status of this feature.


test binary_name::gated ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 3350 filtered out; finished in 1.21s
```

</details>


<details><summary>Can't auto fix this with <code>SNAPSHOTS=overwrite cargo test --test testsuite help::help</code></summary>

```diff
commit cbaa4a89e60507366401ac0076d652f5eafc4998
Author: Lawrence Chou <choznerol@protonmail.com>
Date:   Sat Jun 15 14:36:42 2024 +0800

    Tmp commit prior to `SNAPSHOTS=overwrite cargo t --test testsuite help::`

diff --git a/tests/testsuite/help.rs b/tests/testsuite/help.rs
index 80d236034..c1d8383a2 100644
--- a/tests/testsuite/help.rs
+++ b/tests/testsuite/help.rs
@@ -3,6 +3,7 @@
 #![allow(deprecated)]
 
 use cargo_test_support::registry::Package;
+use cargo_test_support::str;
 use cargo_test_support::{basic_manifest, cargo_exe, cargo_process, paths, process, project};
 use std::fs;
 use std::path::Path;
@@ -17,9 +18,11 @@ fn help() {
     cargo_process("build -h").run();
     cargo_process("help help").run();
     // Ensure that help output goes to stdout, not stderr.
-    cargo_process("search --help").with_stderr("").run();
+    cargo_process("search --help").with_stderr(str![]).run();
     cargo_process("search --help")
-        .with_stdout_contains("[..] --frozen [..]")
+        .with_stdout_contains(str![[r#"
+HEY SNAPBOX, FIX THIS AUTOMATICALLY PLEASE
+"#]])
         .run();
 }
```

```
$ SNAPSHOTS=overwrite cargo test --test testsuite help::help
   Compiling cargo v0.82.0 (/Users/ychou/choznerol/contrib/cargo)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 6.23s
     Running tests/testsuite/main.rs (target/debug/deps/testsuite-00971f03e698357e)

running 5 tests
test cargo_help::help::case ... ok
test help::help_alias ... ok
test help::help ... FAILED
test help::help_external_subcommand ... ok
test help::help_man ... ok

failures:

---- help::help stdout ----
running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo`
running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo help`
running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo -h`
running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo help build`
running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo build -h`
running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo help help`
running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo search --help`
thread 'help::help' panicked at tests/testsuite/help.rs:21:56:

test failed running `/Users/ychou/choznerol/contrib/cargo/target/debug/cargo search --help`
error: stderr did not match:
1        -tests/testsuite/help.rs:21:48


other output:
Search packages in the registry. Default registry is crates.io

Usage: cargo search [OPTIONS] [QUERY]...

Arguments:
  [QUERY]...

Options:
      --limit <LIMIT>        Limit the number of results (default: 10, max: 100)
      --index <INDEX>        Registry index URL to search packages in
      --registry <REGISTRY>  Registry to search packages in
  -v, --verbose...           Use verbose output (-vv very verbose/build.rs output)
  -q, --quiet                Do not print cargo log messages
      --color <WHEN>         Coloring: auto, always, never
      --config <KEY=VALUE>   Override a configuration value
  -Z <FLAG>                  Unstable (nightly-only) flags to Cargo, see 'cargo -Z help' for details
  -h, --help                 Print help

Manifest Options:
      --locked   Assert that `Cargo.lock` will remain unchanged
      --offline  Run without accessing the network
      --frozen   Equivalent to specifying both --locked and --offline

Run `cargo help search` for more detailed information.


note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    help::help

test result: FAILED. 4 passed; 1 failed; 0 ignored; 0 measured; 3346 filtered out; finished in 2.80s

error: test failed, to rerun pass `--test testsuite`
```

</details>


I initially suspected that `help.rs` mostly uses `cargo_process("...")` instead of `p.cargo("...")`, but couldn't really trace the root cause.

<!-- I thought the `SNAPSHOTS` env only affect [`assert_ui`](https://github.com/choznerol/cargo/blob/reproduce-run_expect_error-with_stderr_contains-incompatibility/crates/cargo-test-support/src/compare.rs#L93) and [`assert_e2e`](https://github.com/choznerol/cargo/blob/reproduce-run_expect_error-with_stderr_contains-incompatibility/crates/cargo-test-support/src/compare.rs#L135). -->

Reply:
Note that this specific test was later removed in https://github.com/rust-lang/cargo/pull/14060#discussion_r1640859247

Reply:
Thanks! Just get `SNAPSHOTS=overwrite` working and pushed https://github.com/rust-lang/cargo/commit/8db30bc4bdd67de9947f1f9a5dd653edcc2c8d59.